core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -211,7 +211,7 @@ private[spark] class BlockManager(
    *
    * @param blockSize the decrypted size of the block
    */
-  private abstract class BlockStoreUpdater[T](
+  private[spark] abstract class BlockStoreUpdater[T](
```

Comment:
I think you only need to change scope of TempFileBasedBlockStoreUpdater, not BlockStoreUpdater?

Reply:
I'm afraid not. if we revive the scope here, compiler would say:

```
[error] /Users/wuyi/workspace/spark/core/src/main/scala/org/apache/spark/storage/BlockManager.scala:368: private class BlockStoreUpdater escapes its defining scope as part of type BlockManager.this.BlockStoreUpdater[T]
[error]     extends BlockStoreUpdater[T](blockSize, blockId, level, classTag, tellMaster, keepReadLock) {
[error]             ^
[error] /Users/wuyi/workspace/spark/core/src/main/scala/org/apache/spark/storage/BlockManager.scala:368: no arguments allowed for nullary constructor Object: ()Object
[error]     extends BlockStoreUpdater[T](blockSize, blockId, level, classTag, tellMaster, keepReadLock) {
[error]                                  ^
[error] /Users/wuyi/workspace/spark/core/src/main/scala/org/apache/spark/storage/BlockManager.scala:383: value save is not a member of Product with Serializable
[error]       val res = super.save()
[error]                       ^
[error] three errors found
[error] Compile failed at 2019-7-31 23:08:25 [36.500s]
```